src/replicache.ts

```diff
@@ -447,6 +466,42 @@ export class Replicache<MD extends MutatorDefs = {}> {
     resolveLicenseCheck(true);
   }
 
+  private async _startLicenseActive(
+    resolveLicenseActive: (valid: boolean) => void,
+    lc: LogContext,
+  ): Promise<() => void> {
+    if (
+      this._licenseKey === undefined ||
+      this._licenseKey === TEST_LICENSE_KEY
+    ) {
+      resolveLicenseActive(false);
+      return noop;
+    }
+
+    const activate = async () => {
+      try {
+        // TODO(phritz) add browser profile
+        await licenseActive(
+          browserSimpleFetch,
+          PROD_LICENSE_SERVER_URL,
+          this._licenseKey as string,
+          'TODO-BROWSER-PROFILE-ID',
+        );
+      } catch (err) {
+        this._logger.info?.(`Error sending license active ping: ${err}`);
+      }
+    };
+    await activate();
+    resolveLicenseActive(true);
```

Comment:
Whether the license is *active* is only a function of the license key which is known at construction time. Why use the promise/resolver mechanism to delay computing that? Seems more straightforward to compute it in the getter based on the license key.